app/controllers/api/v0/maintainers_controller.rb

```diff
@@ -47,7 +47,7 @@ def create
           conditions = data.select {|key, _| valid_fields.include?(key)}
           begin
             maintainer = Maintainer.find_or_create_by(conditions)
-          rescue ActiveRecord::RecordNotUnique
+          rescue ActiveRecord::RecordNotFound
```

Comment:
The behavior of `find_or_create_by` changed in 7.1, and it’s now more like `find_or_create_or_find_by`. (https://github.com/rails/rails/issues/51149) Kinda weird. Maybe want to modify how we do this operation to work differently now for efficiency’s sake.

Maybe just:

```rb
maintainer = Maintainer.find_by(conditions) || Maintainer.create!(conditions)
```

Or be more complicated, and find by name, then if `parent_uuid` is set, assert that it matches. 🤷 

Either way, Rails core folks don’t view our usage here as a pattern they intend to support (it *is* probably unusual).